### PR TITLE
this commit adds the functionality to view all the comments on an art…

### PR DESCRIPTION
**Pull request by cheesenbeans**
…icle page

### CODE REVIEW

File: src/components/SingleArticle.jsx [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
     
     useEffect(()=>{
         fetchArticleById(article_id).then((article)=>{
             setSingleArticle(article)
             setIsLoading(false)
+            return article
+        }).then((article)=> {
+            fetchCommentsByArticleId(article.article_id).then((comments)=>{
+                setCommentsByArticleId(comments)
+            })
```
Comment (aloush-dev) [inline src/components/SingleArticle.jsx]: This wouldn't necessarily need to be in a then chain as you have access to the article id, and this has lead to nested then blocks.

File: src/components/SingleArticle.jsx [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
             <h4>Written by: {singleArticle.author}</h4>
             <h4>Topic: {singleArticle.topic}</h4>
             <p>Votes: {singleArticle.votes}</p>
             <p>{singleArticle.body}</p>
+            </div>
+            <section className="comments">
+                <h3 className="commentsh3">Comments</h3>
+                    <ul>
+                    {commentsByArticleId.map((comment) => {
+                            return <CommentCard key={comment.comment_id} comment={comment} />;
```
Comment (aloush-dev) [inline src/components/SingleArticle.jsx]: nice job seperating commentCard into a new component

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Nice job on this PR.

Think about what to display to the user if an article doesn't have any comments.
Also About what will display while the comments are still loading

**Review by Slyose — APPROVED** [review]
Nice implementation of changes! Go ahead and merge when ready ✅

**Pull request merged**